pyanaconda/ui/tui/spokes/software_selection.py

```diff
@@ -223,31 +224,32 @@ def refresh(self, args=None):
             else:
                 msg = _("No add-ons to select.")
 
-            for addon_id in args:
+            for i, addon_id in enumerate(args):
                 name = self.payload.groupDescription(addon_id)[0]
                 selected = addon_id in self._addons_selection
-                displayed.append(CheckboxWidget(title="%s" % name, completed=selected))
-
-        def _prep(i, w):
-            """ Do some format magic for display. """
-            num = TextWidget("%2d)" % (i + 1))
-            return ColumnWidget([(4, [num]), (None, [w])], 1)
+                widget = CheckboxWidget(title="%s" % name, completed=selected)
+                self._container.add(widget, callback=self._set_addons_callback, data=(i, args))
```

Comment:
You can pass data instead of the index. 

Just a note. If you pass the index, it means that we cannot modify the "choices" till user's input. However, with DBus it could be sometimes useful to be connected to properties-changed signal and update the choices with every received signal. If we pass data, we can do it this way.